### PR TITLE
Fix "forbidden" Error From AWS

### PR DESCRIPTION
**Pull request by kkriehl**
The AWS SDK and Caddy Go dependencies have been updated to the latest versions. This fixes an (unknown) error with the generated signature that resulted in a `forbidden` response from AWS.

The Go version has also been updated to comply with the version that is used in the `caddy:2.7-builder` Docker image.

With these changes the module works as intended when built with xcaddy v0.4.1 and within the Docker image, at least for my use case: accessing a route in AWS API Gateway that is protected by the IAM authorizer. I would assume that the module works for other AWS Sigv4 protected endpoints as well.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
-go 1.20
+go 1.21.10
 
 require (
 	github.com/BishopFox/aws-signing v1.0.1
-	github.com/aws/aws-sdk-go-v2 v1.2.0
-	github.com/caddyserver/caddy/v2 v2.7.0
+	github.com/aws/aws-sdk-go-v2 v1.26.1
+	github.com/caddyserver/caddy/v2 v2.7.6
```
Comment (mohammed90) [inline go.mod]: Can you keep Caddy at v2.7.0? It'll allow builds of Caddy across all versions from v2.7.0 and higher.

**Review by mohammed90 — CHANGES_REQUESTED** [review]
Thank you! The aws SDK has quite the jump 😮 Just one minor thing and I'll merge

File: go.mod (hunk 1 above)
Comment (kkriehl) [reply go.mod]: Good point. I'll check everything again (just in case) and push an update.

Comment (kkriehl) [reply go.mod]: @mohammed90 I've replaced the commit with an update that retains the caddy version.

**Review by mohammed90 — APPROVED** [review]

**Pull request merged**